sycl/test/lit.cfg.py

```diff
@@ -200,13 +203,24 @@ def getDeviceCount(device_type):
     lit_config.warning("Accelerator device not found")
 
 if xocc != "off":
-    xrt_lock="/tmp/xrt.lock"
-    acc_run_substitute+= "setsid flock -x " + xrt_lock + " "
+    # xrt doesn't deal well with multiple executable using it concurrently(at the time of writing).
+    # the detail are at https://xilinx.github.io/XRT/master/html/multiprocess.html
+    # so we wrap every use of XRT inside an file lock
+    # we also wrap invocation of executable in an setsid to prevent
+    # a single program failure from ending all the tests.
+    xrt_lock = "/tmp/xrt.lock"
+    acc_run_substitute+= "setsid flock --exclusive " + xrt_lock + " "
     if os.path.exists(xrt_lock):
         os.remove(xrt_lock)
-    if "XCL_EMULATION_MODE" in os.environ:
-        if os.environ["XCL_EMULATION_MODE"] == "hw":
-            acc_run_substitute="env -u XCL_EMULATION_MODE " + acc_run_substitute
+    # XCL_EMULATION_MODE = hw is only valid for our SYCL driver, 
+    # for XRT XCL_EMULATION_MODE should only be used when using either hw_emu or sw_emu
+    if xocc_target == "hw":
+        acc_run_substitute="env -u XCL_EMULATION_MODE " + acc_run_substitute
```

Comment:
```suggestion
        acc_run_substitute="env --unset=XCL_EMULATION_MODE " + acc_run_substitute
```
Explicit programming